keps/sig-apps/3850-backoff-limits-per-index-for-indexed-jobs/README.md

```diff
@@ -0,0 +1,1266 @@
+However, the `backoffLimitPerIndex` field not only allows to count failures
+towards the backoff limit per index, but also allows all indexes to execute
+despite failures, thus more generic names, like (1.) and (2.) are preferred.
+
+Also the options (1.) and (2.) may be reused in the context of success policy
+which is subject of
+[Job success/completion policy](https://github.com/kubernetes/kubernetes/issues/117600).
+It might be beneficial for the API to consider the conditions for the Job
+success or failure under the same field.
+
+### Mark Job Complete if some indexes failed
+
+The alternative to the proposed [Job completion](#job-completion) strategy.
+
+Allow execution of all indexes, up to `.spec.runPolicy.maxFailedIndexes` of
+failed indexes. Then, mark the Job `Complete` even if some indexes failed.
+The Job is marked `Failed` only if the number of failed indexes exceeds the
+specified `.spec.runPolicy.maxFailedIndexes` limit, in that case, the `reason`
+field could be `FailedIndexes`, and the `message` field would list the failed
+indexes up to a couple of them.
+
+**Reasons for deferring / rejecting**
+
+This approach is less intuitive to the end-users of the API, compared
+to the proposal. In particular, in some cases it would require custom logic in
+the user's controller to determine if the Job is failed.
+
+### Support backoffLimitPerIndex when restartPolicy=OnFailure
+
+We've considered supporting the backoffLimitPerIndex when pod's `restartPolicy=OnFailure`.
+However, in this mode it is Kubelet's responsibility to restart the pod, while
+the maximal number of restarts would be enforced by the Job controller, which
+is racy in its nature. For example, in-between the checks by the Job controller,
+Kubelet execute more restarts than the specified `.spec.backoffLimit`.
+
+**Reasons for deferring / rejecting**
+
+We believe that this feature can be supported well by using the pod-level API,
+started in this KEP:
+[Add a new field maxRestartTimes to podSpec when running into RestartPolicyOnFailure](https://github.com/kubernetes/enhancements/issues/3322).
+
+Once the pod-level API is done, it could be considered to support `.spec.runPolicy.backoffLimitPerIndex`
+when`restartPolicy=OnFailure` in pod's spec. In this case we could set the pod-level
+`maxRestartTimes` field based on the Job-level `.spec.backoffLimit`, leaving the
+responsibility of enforcing the limit to the Kubelet.
+
+### Set backoffLimitPerIndex at the top level
+
+We could set the backoffLimitPerIndex at top-level in order to put it next to
+the backoffLimit field. For example:
+
+```yaml
+apiVersion: v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  backoffLimit: 4
+  backoffLimitPerIndex: true
+  ...
+```
+
+**Reasons for deferring / rejecting**
+
+The backoffLimitPerIndex is also conceptually close to maxFailedIndexes.
+We will consider this option based on the community feedback.
```

Comment:
See my other comment on the topic, would be good to get feedback from api reviewers about it.

Reply:
+1 from me, @deads2k @alculquicondor wdyt? 

Maybe we can have something like runPolicy in Job v2, and for now just put the new fields top level, next to backoffLimit?